proto/redpanda/api/dataplane/v1alpha1/secret.proto

```diff
@@ -0,0 +1,191 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+// Secret defienes the secret resource
+message Secret {
+  string id = 1 [
+    (google.api.field_behavior) = IMMUTABLE,
+    (google.api.field_behavior) = OUTPUT_ONLY
+  ];
+  map<string, string> labels = 2 [
+    (buf.validate.field).map = {
+      values: {
+        string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+      },
+    },
+    (google.api.field_behavior) = IMMUTABLE
+  ];
+}
+
+message SecretInput {
+  string id = 1 [(google.api.field_behavior) = IMMUTABLE];
+  map<string, string> labels = 2 [
+    (buf.validate.field).map = {
+      values: {
+        string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+      },
+    },
+    (google.api.field_behavior) = IMMUTABLE
+  ];
+  repeated bytes secret_data = 3 [
+    (google.api.field_behavior) = INPUT_ONLY,
+    (google.api.field_behavior) = REQUIRED
+  ];
+}
+
+// ListSecretsResponse is the response of ListSecrets
+message ListSecretsResponse {
+  repeated Secret secrets = 1;
+  // token to retrieve the next page.
+  string next_page_token = 2;
+}
+
+// ListSecretsRequest is the request of ListSecrets
+message ListSecretsRequest {
+  // Defines the filter for Secrets
+  message Filter {
+    string name = 1;
+    map<string, string> labels = 2 [(buf.validate.field).map = {
+      values: {
+        string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+      },
+    }];
+  }
+  Filter filter = 1;
+  // Value of the next_page_token field returned by the previous response. If not provided, the system assumes the first page is requested.
+  string page_token = 2;
+  // Limit the paginated response to a number of items.
+  int32 page_size = 3 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+    description: "Limit the paginated response to a number of items.",
+    minimum: 1,
+    maximum: 50
+  }];
```

Comment:
should we add identical buf validation annotations?

Reply:
Same as above, the protos for anything but user.proto are just drafts for the OpenAPI spec.